db/active_replicator_common_collections.go

```diff
@@ -30,26 +31,43 @@ func (arc *activeReplicatorCommon) _initCollections() ([]replicationCheckpoint,
 	}
 
 	if arc.config.CollectionsLocal != nil {
-		for _, scopeAndCollectionName := range arc.config.CollectionsLocal {
-			scopeName, collectionName, err := parseScopeAndCollection(scopeAndCollectionName)
+		for i, localScopeAndCollection := range arc.config.CollectionsLocal {
+			localScopeName, localCollectionName, err := parseScopeAndCollection(localScopeAndCollection)
 			if err != nil {
 				return nil, err
-			} else if scopeName == nil || collectionName == nil {
-				return nil, fmt.Errorf("scope and collection name must be specified: %q - %v", scopeAndCollectionName, arc.config.CollectionsLocal)
+			} else if localScopeName == nil || localCollectionName == nil {
+				return nil, fmt.Errorf("scope and collection name must be specified: %q - %v", localScopeAndCollection, arc.config.CollectionsLocal)
+			}
```

Comment:
I think this would read shorter if was a function:

```func getScopeAndCollectionName(scopeName, collectionName) (base.ScopeAndCollectionName, error)```